.github/workflows/docgen.yml

```
@@ -0,0 +1,43 @@
+name: Nightly
+
+on:
+  schedule:
+      - cron:  '0 12 * * *'
+  workflow_dispatch:
+
+jobs:
+  docgen:
+    name: Generate Reference Docs
+    if: github.repository_owner == 'withastro'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code using Git
+        uses: actions/checkout@v3
+
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v2.2.1
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'pnpm'
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Run docgen script
+        run: pnpm run docgen
+
+      - name: Create Pull Request
+        id: createpr
+        uses: peter-evans/create-pull-request@v3
+        with:
+          branch: ci/docgen
+          token: ${{ secrets.NIGHTLY_PERSONAL_GITHUB_TOKEN }}
+          add-paths: src/pages/en/*.md
```

The glob value seems a bit broad for my taste, considering that docgen only writes a single file.

What do you think about limiting this to `src/pages/en/reference/*.md` or even the specific `configuration-reference.md` file inside that directory?

---

I wanted to keep it slightly broader than just the specific file to save us needing to modify this if/when we add CLI reference to the docgen script as I figured we’d very likely forget to update this. But I think scoping to the reference directory is probably fair.

```suggestion
          add-paths: src/pages/en/reference/*.md
```